pkg/kubelet/kubelet.go

```diff
@@ -901,6 +901,23 @@ func (kl *Kubelet) initialNodeStatus() (*api.Node, error) {
 		if err != nil {
 			return nil, err
 		}
+
+		// If the cloud has zone information, label the node with the zone information
+		zones, ok := kl.cloud.Zones()
```

Comment:
We could do that, but a few reasons why I'd rather not:

* This means that the node comes up with the zone information, avoiding a window where clients would have to deal with the zone information not yet being available (and differentiating between that case vs the case where the helper pods isn't configured)
* It's pretty easy to do it here in the kubelet - we're already creating the node, we already have the cloudprovider spun up -  so I do wonder if it is worth the complexity of creating a separate helper program & podifying it & ensuring it is configured correctly etc.
* Having this information reliably on the nodes lets people target it confidently.  I want to use it for Ubernetes-Lite, where we will use this information to have the scheduler spread pods in an RC across zones when a cluster is running in multiple-AZs.  Ubernetes-Full will have a richer model, but I do think Uberentes-Lite will be in use for a while yet.